programs/disks/CommandRemove.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "ICommand.h"
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+class CommandRemove : public ICommand
+{
+public:
+    CommandRemove() = default;
+    CommandRemove(std::optional<ProgramOptionsDescription> & command_option_description_input)
+    {
+        command_name = "remove";
+        command_option_description.emplace(createOptionsDescription("Help Message for remove", getTerminalWidth()));
+        description = "Remove file or directory with all children. Throws exception if file doesn't exists.";
+        usage = "Usage: remove [OPTION]... <PATH>";
+        command_option_description->add(command_option_description_input.value());
+        command_option_description->add_options()
+            ("help,h", "print help message for remove");
+    }
+
+    void execute(
+        const std::vector<String> & command_arguments,
+        const DB::ContextMutablePtr & global_context,
+        const Poco::Util::AbstractConfiguration & config) override
+    {
+        if (command_arguments.size() > 1)
+        {
+            printHelpMessage();
+            throw DB::Exception("Bad Arguments", DB::ErrorCodes::BAD_ARGUMENTS);
+        }
+
+        String disk_name = config.getString("disk", "default");
+
+        String path = (command_arguments.size() < 1
+                                ? ""
+                                : command_arguments[0]);
+
+        DiskPtr disk = global_context->getDisk(disk_name);
+
+        String full_path = fs::path(disk->getPath()) / path;
```

Comment:
Let's protect against forbidden paths. Need to check that `(fs::absolute(disk->getPath()) / path).lexically_normal()` is indeed inside `disk->getPath()`

Need to add this path validation in all places it is used (move checking into a separate method `validatePath`).